mule-extensions-api/src/main/java/org/mule/runtime/extension/api/notification/NotificationActionDefinition.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.notification;
+
+/**
+ * Identifies a notification action to be declared by extensions and must be implemented by an {@link Enum}.
+ *
+ * @param <E> Enum type
+ * @since 1.1
+ */
+public interface NotificationActionDefinition<E extends Enum<E>> {
+
+  /**
+   * Indicates the {@link Class} of the data associated to the notification
+   *
+   * @return the type of the notification data.
+   */
+  Class<?> getDataType();
```

Comment:
why class instead of an actual DataType?

Reply:
`DataType` does not get along well with `MetadataType`.

Reply:
why not? you can use the inner class to build the metadatatype and use the mimeType to infer a MetadataFormat. Plus, there are plans to further enrich DataType in the future with things that MetadataType can definitively use

Reply:
The first part is trivial but the latter not so much: all TypeBuilders require the MetadataFormat first and TypeLoaders hard code it. It would require either introducing a new TypeLoader, modifying the Java one or some other thing.